src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -47,6 +47,23 @@ class AgoraBusiness {
     }
   }
 
+  def delete(agoraEntity: AgoraEntity, entityTypes: Seq[AgoraEntityType.EntityType], username: String): Int = {
+    if (!NamespacePermissionsClient.getNamespacePermission(agoraEntity, username).canRedact)
+      throw new NamespaceAuthorizationException(AgoraPermissions(Redact), agoraEntity, username)
+
+    // if the entity was a method, then redact all associated configurations
+    if (entityTypes equals AgoraEntityType.MethodTypes) {
+      val entityWithId = AgoraDao.createAgoraDao(entityTypes)
+        .findSingle(agoraEntity.namespace.get, agoraEntity.name.get, agoraEntity.snapshotId.get)
+
+      val configurations = AgoraDao.createAgoraDao(entityTypes).findConfigurations(entityWithId.id.get)
```

Comment:
Might be slightly more efficient to create the agora Dao once and re-use it later?
val agoraDao = AgoraDao.createAgoraDao(entityTypes)
agoraDao.find...